pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -42,52 +44,97 @@ def __call__(self, time: float):
-                # occurs in this timestep between the given
-                # persons.
-                r = random.random()
-
-                if r < force_of_infection:
-                    infectee_cell.enqueue_person(infectee)
+            if Parameters.instance().do_CovidSim:
+                # Chooses cells based on a cumulative transmission array
+                # one and a time a tests each infection event.
+                while number_to_infect > 0:
+                    # Weighting for cell choice in Covidsim uses cum_trans and
+                    # invCDF arrays, but really can't see how these are
+                    # initialised. Have used the number of susceptible for now.
+                    weights = [cell2.compartment_counter.retrieve()
+                               [InfectionStatus.Susceptible]
+                               for cell2 in possible_infectee_cells]
+                    infectee_cell = random.choices(possible_infectee_cells,
+                                                   weights=weights, k=1)[0]
+                    # Sample at random from the infectee cell to find
+                    # an infectee
+                    infectee = random.sample(infectee_cell.persons, 1)[0]
+                    infection_distance = DistanceFunctions.dist_euclid(
+                     cell.location, infectee_cell.location) / Parameters.\
+                        instance().infection_radius
+                    if (infection_distance < random.random()):
+                        # Covidsim rejects the infection event if the distance
+                        # between infector/infectee is too large.
+                        self.do_infection_event(infector, infectee, timestep)
+                        number_to_infect -= 1
+
+            else:
+                # Chooses a list of cells (with replacement) for each infection
+                # event to occur in. Specifically inter-cell infections
+                # so can't be the same cell
+
+                distance_weights = [1/DistanceFunctions.dist_euclid(
+                                    cell.location, cell2.location)
+                                    for cell2 in
+                                    possible_infectee_cells]
+                cell_list = random.choices(possible_infectee_cells,
+                                           weights=distance_weights,
+                                           k=number_to_infect)
+                print(cell_list)
+
+                # Each infection event corresponds to a infectee cell
+                # on the cell list
+                for infectee_cell in cell_list:
+
+                    # Sample at random from the infectee cell to find
+                    # an infectee
+                    infectee = random.sample(infectee_cell.persons, 1)[0]
+                    self.do_infection_event(infector, infectee, timestep)
+
+    def do_infection_event(self, infector, infectee,
```

Comment:
If this method isn't private (and maybe even if it is) it should be unit tested - not that I'm trying to get my tests re-added ;)

Reply:
Hah, okey, I can separate that out